lib/core/regs/consent.ts

```diff
@@ -0,0 +1,134 @@
+import { inferRegulation, Regulation } from "./regulations";
+import { PingReturn as GPPConsentData } from "./gpp/cmpapi";
+import { TCData as TCFConsentData } from "./tcf/cmpapi";
+import { SectionID as TCFEuV2SectionID, APIPrefix as TCFEuV2APIPrefix } from "./gpp/tcfeuv2";
+import { SectionID as TCFCaV1SectionID } from "./gpp/tcfcav1";
+
+type Consent = {
+  deviceAccess: boolean;
+  reg: Regulation | null;
+  tcf?: string;
+  gpp?: string;
+};
+
+function gdprConsent(): Consent {
+  const consent: Consent = { deviceAccess: false, reg: "gdpr" };
+
+  // For use TCF if available, otherwise use GPP,
+  // if none available assume device access is not allowed
+  if (hasTCF()) {
+    onTCFChange((data) => {
+      consent.deviceAccess = tcfDeviceAccess(data);
+      consent.tcf = data.tcString;
+    });
+  } else if (hasGPP()) {
+    onGPPSectionChange(TCFEuV2SectionID, (data) => {
+      consent.deviceAccess = gppEUDeviceAccess(data);
+      consent.gpp = data.gppString;
+    });
+  }
+
+  return consent;
+}
+
+function usConsent(): Consent {
+  return { deviceAccess: true, reg: "us" };
+}
+
+function canConsent(): Consent {
+  const consent: Consent = { deviceAccess: true, reg: "can" };
+  onGPPSectionChange(TCFCaV1SectionID, (data) => {
+    // Device access is always granted
+    consent.gpp = data.gppString;
+  });
+
+  return consent;
+}
+
+function getConsent(reg: Regulation | null): Consent {
+  switch (reg) {
+    case "gdpr":
+      return gdprConsent();
+    case "us":
+      return usConsent();
+    case "can":
+      return canConsent();
+    default:
+      return { deviceAccess: true, reg: null };
+  }
+}
+
+function gppEUDeviceAccess(data: GPPConsentData): boolean {
+  if (!(TCFEuV2APIPrefix in data.parsedSections)) {
+    return false;
+  }
+
+  const section = data.parsedSections[TCFEuV2APIPrefix] || [];
+  const publisherSubsection = section.find((s) => {
+    return "SegmentType" in s && s.SegmentType === 3;
+  });
+
+  if (!publisherSubsection) {
+    return false;
+  }
+
+  return (
+    publisherSubsection.PubPurposesConsent.includes(1) || publisherSubsection.PubPurposesLITransparency.includes(1)
+  );
+}
+
+function tcfDeviceAccess(data: TCFConsentData): boolean {
+  if (!data.gdprApplies) {
+    return true;
+  }
+  return !!data.publisher.consents["1"] || !!data.publisher.legitimateInterests["1"];
```

Comment:
this is incorrect, one cannot have LI on purpose 1. 
You could check data.purpose.consents and a vendor id you take as config or data.publisher.consents["1"].

Going to a friendly site, https://highfivve.com/en/from-publisher-to-publisher/, and typing 
``` 
__tcfapi('addEventListener', 2, function(tcData, success) { console.log('cmp responded:', tcData, success);});
```
your check fails bc no publisher purposes are defined and you aren't checking data.purpose at all. To check vendors, you could look at data.vendor.consents, customVendorConsents, and/or googleVendorConsents. I recommend optable register for that final list

Reply:
Thanks, I'll remove LI purpose 1 check.
As for checking vendor consent it's a good idea and I'll go ahead with checking `vendor.consents` alongside `purpose.consents` if a vendor ID is provided.

For `customVendorConsents` and `googleVendorConsents` properties I'm having trouble finding where they come from, I don't see it documented in https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata . Is it possible it's a consentmanager.net only thing? The only ref I found is https://help.consentmanager.net/books/cmp/page/changes-to-the-iab-cmp-framework-js-api

Reply:
It is documented here

https://support.google.com/admanager/answer/9681920?hl=en

Reply:
Might have to parse the ac string to get the additional consented vendor list. I do suggest Optable register for this list, then you could just check to consent to yourself